### PR TITLE
Exclude "potentially dangerous function" check.

### PR DESCRIPTION
**Pull request by kevingranade**
SUMMARY: None

Just some LGTM cleanup, there's nothing to be done about these functions and they aren't dangerous in our use case.

I'm mostly opening as a PR because I'm not 100% sure abut the LGTM yaml.

### CODE REVIEW

File: .lgtm.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # Disable FIXME query, in DDA it is as likely to be a future feature as a bug.
+# Disable "potentially dangerous function" warning like localtime, there is no good alternative and we aren't multithreaded.
 queries:
-  - exclude: cpp/fixme-comment
-  
+  - exclude:
+      cpp/fixme-comment
+      cpp/potentially-dangerous-function
```
Comment (kevingranade) [inline .lgtm.yml:6]:
```suggestion
      - cpp/fixme-comment
      - cpp/potentially-dangerous-function
```

File: .lgtm.yml [hunk 2]
```diff
@@ -1,4 +1,6 @@
 # Disable FIXME query, in DDA it is as likely to be a future feature as a bug.
+# Disable "potentially dangerous function" warning like localtime, there is no good alternative and we aren't multithreaded.
 queries:
-  - exclude: cpp/fixme-comment
-  
+  - exclude:
+      - cpp/fixme-comment
+      - cpp/potentially-dangerous-function
```
Comment (kevingranade) [inline .lgtm.yml:6]:
```suggestion
  - exclude:
      id:
          - cpp/fixme-comment
          - cpp/potentially-dangerous-function
```

File: .lgtm.yml [hunk 3]
```diff
@@ -1,4 +1,7 @@
 # Disable FIXME query, in DDA it is as likely to be a future feature as a bug.
+# Disable "potentially dangerous function" warning like localtime, there is no good alternative and we aren't multithreaded.
 queries:
-  - exclude: cpp/fixme-comment
-  
+  - exclude:
+      id:
+          - cpp/fixme-comment
+          - cpp/potentially-dangerous-function
```
Comment (kevingranade) [inline .lgtm.yml:7]:
```suggestion
  - exclude: cpp/fixme-comment
  - exclude: cpp/potentially-dangerous-function
```

File: .lgtm.yml [hunk 4]
```diff
@@ -1,4 +1,5 @@
 # Disable FIXME query, in DDA it is as likely to be a future feature as a bug.
+# Disable "potentially dangerous function" warning like localtime, there is no good alternative and we aren't multithreaded.
 queries:
   - exclude: cpp/fixme-comment
-  
+  - exclude: cpp/potentially-dangerous-function
```
Comment (kevingranade) [inline .lgtm.yml:5]:
```suggestion
  - exclude: cpp/potentially-dangerous-function
  
```

**Comment by ifreund** [comment]
Hmm, looks like LGTM didn't run on the latest commit for some reason

**Comment by kevingranade** [comment]
Yea neither the final nor the penultimate version.

**Pull request closed without merging**